zips/zip-0246.rst

```diff
@@ -32,41 +32,58 @@ field, as specified in section 7.1 of the Zcash protocol specification
 Abstract
 ========
 
-This ZIP defines the digest algorithms associated with the v6 transaction format.
-
-This proposal also defines the new concept of "sighash versioning": where previously
-each transaction version had a single associated digest algorithm, going forward it
-will be possible for signers to use any digest algorithm within the closed set
-specified for a given transaction version (and made available in consensus via network
-upgrades).
+This ZIP defines the digest algorithms associated with the v6 transaction
+format.
 
+This proposal also defines the new concept of "sighash algorithm versioning":
+where previously each transaction version had a single associated digest
+algorithm, going forward it will be possible for signers to use any digest
+algorithm within the closed set specified for a given transaction version (and
+made available in consensus via network upgrades).
 
 ==========
 Motivation
 ==========
 
-TBD
-
-TODO: Motivate sighash versioning
+The motivation for sighash algorithm versioning is that we want to be able to
+specify the V6 transaction format and signatures without knowing what the
+signatures for atomic swaps in action bundles will need to commit to. This
+allows for future signature hash algorithms to be specified for V6 transactions
+that support signing parts of a transaction while leaving the remainder of the
+transaction malleable, or that commit to additional state that nodes may start
+tracking in a future upgrade that is not part of an individual transaction
+being signed.
 
+Being able to remove support for sighash algorithm versions makes it possible
+to respond effectively in the case that a bug is found in a previously-deployed
+sighash algorithm.
 
 ============
 Requirements
 ============
 
-- Continue to support existing functionality of the protocol (multisig,
-  signing modes for transparent inputs).
+- Continue to support existing functionality of the protocol (multisig, signing
+  modes for transparent inputs).
+
+- It should be possible to update this ZIP with additional sighash algorithm
+  versions that might have divergent behavior with respect to previous
+  versions, after previous versions have been deployed in network upgrades.
 
-- It should be possible to update this ZIP with additional digest versions after the first
-  version has been deployed in a network upgrade.
+- It should be possible to remove support for sighash algorithm versions in
```

Comment:
```suggestion
- It should be possible to disable support for sighash algorithm versions in
```
Subtly different because the support can't be immediately removed from implementations; it is still needed to check consensus before the upgrade.